python/example/brunel.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+import arbor
+import numpy,argparse
+
+'''
+A Brunel network consists of nexc excitatory LIF neurons and ninh inhibitory LIF neurons.
+Each neuron in the network receives in_degree_prop * nexc excitatory connections
+chosen randomly, in_degree_prop * ninh inhibitory connections and next (external) Poisson connections.
+All the connections have the same delay. The strenght of excitatory and Poisson connections is given by
+parameter weight, whereas the strength of inhibitory connections is rel_inh_strength * weight.
+Poisson neurons all spike independently with expected number of spikes given by parameter poiss_lambda.
+Because of the refractory period, the activity is mostly driven by Poisson neurons and
+recurrent connections have a small effect.
+
+Call with parameters, for example:
+./brunel.py -n 400 -m 100 -e 20 -p 0.1 -w 1.2 -d 1 -g 0.5 -l 5 -t 100 -s 1 -G 50 -S 123 -f spikes.txt
+
+'''
+
+# Samples m unique values in interval [start, end) - gid.
+# We exclude gid because we don't want self-loops.
+def sample_subset(gid, start, end, m):
+    gen = numpy.random.default_rng(gid+42)
+    s = set()
+    while len(s) < m:
+        val = gen.integers(low=start,high=end - 1)
+        if val != gid:
+            s.add(val)
+    return s
+
+class brunel_recipe (arbor.recipe):
+    def __init__(self, nexc, ninh, next, in_degree_prop, weight, delay, rel_inh_strength, poiss_lambda, seed = 42):
+
+        arbor.recipe.__init__(self)
+        # self.props = arbor.neuron_cable_properties()
+        # self.cat = arbor.default_catalogue()
+        # self.props.register(self.cat)
+
+        # Make sure that in_degree_prop in the interval (0, 1]
+        if 0.0>=in_degree_prop>1.0:
```

Comment:
This test is always false (and indeed, something like `-p 1.1` gives us an infinite loop.)

Reply:
Whoops!